js/client.js

```diff
@@ -1636,19 +1650,37 @@
 			document.title = room.title ? room.title + " - Showdown!" : "Showdown!";
 		},
 		updateAutojoin: function () {
-			if (Config.server.id !== 'showdown') return;
+			if (!Config.server.registered) return;
+			var curAutojoin = (Tools.prefs('autojoin') || '');
+			if (typeof curAutojoin === 'string' && Config.server.id !== 'showdown') {
+				// Switch to the autojoin object to handle multiple servers
+				curAutojoin = {showdown: curAutojoin};
+			}
 			var autojoins = [];
 			var autojoinCount = 0;
 			var rooms = this.roomList.concat(this.sideRoomList);
 			for (var i = 0; i < rooms.length; i++) {
 				var room = rooms[i];
 				if (room.type !== 'chat') continue;
+				if (room.id === 'staff' || room.id === 'upperstaff' || (Config.server.id !== 'showdown' && room.id === 'lobby')) continue;
```

Comment:
We still want to store order of Staff and Upper Staff. That's why they're below rather than above the autojoin list.